### PR TITLE
Reduce error pollution even more on post request failures

### PR DESCRIPTION
**Pull request by melroy89**
This will fix the 16th [miscellaneous issue](https://github.com/MbinOrg/mbin/issues/1119), with too much verbose error logging.

I try to handle the post request basically the same way as the previously implemented `InvalidApPostException` exceptions like in the `getCollectionObjectImpl` function.

 
This meaning the content logging will be part of the catch -> `logRequestException` function instead. And the `InvalidApPostException` exception will only log the request body, but no longer the response content. Also we log the status code now as well for failed post requests. Plus some small refactoring.

Last but not least, I will only do a `json_encode` once.

### CODE REVIEW

File: src/Service/ActivityPub/ApHttpClient.php [hunk 1]
```diff
@@ -379,20 +379,28 @@ public function post(string $url, User|Magazine $actor, ?array $body = null): vo
...
 
-        if (!str_starts_with((string) $response->getStatusCode(), '2')) {
-            throw new InvalidApPostException("Post fail: $url, ".substr($response->getContent(false), 0, 1000).' '.json_encode($body));
+            $statusCode = $response->getStatusCode();
+            if (!str_starts_with((string) $statusCode, '2')) {
+                // Do NOT include the response content in the error message, this will be often a full HTML page
+                throw new InvalidApPostException("Post failed: $url, status code: $statusCode, request body: $jsonBody");
+            }
+        } catch (\Exception $e) {
+            $this->logRequestException($response, $url, 'ApHttpClient:post', $e);
```
Comment (melroy89) [inline src/Service/ActivityPub/ApHttpClient.php:403]: Apparently `logRequestException` is logging for "get failed..." .. I will either need to create a `logRequestException` or provide an additional parameter for the message "get failed" or when it's "post failed".

Comment (melroy89) [reply src/Service/ActivityPub/ApHttpClient.php:403]: or .. just remove the word "get"

Comment (melroy89) [reply src/Service/ActivityPub/ApHttpClient.php:413]: Solved.

**Comment by melroy89** [comment]
I'm running this for 4 days without any issues at kbin.melroy.org

**Review by TheVillageGuy — APPROVED** [review]

**Pull request merged**